datasources/src/main/java/org/opensearch/sql/datasources/auth/DataSourceUserAuthorizationHelperImpl.java

```diff
@@ -49,11 +51,12 @@ public void authorizeDataSource(DataSourceMetadata dataSourceMetadata) {
         }
       }
       if (!isAuthorized) {
-        throw new SecurityException(
+        throw new OpenSearchSecurityException(
             String.format(
                 "User is not authorized to access datasource %s. "
                     + "User should be mapped to any of the roles in %s for access.",
-                dataSourceMetadata.getName(), dataSourceMetadata.getAllowedRoles().toString()));
+                dataSourceMetadata.getName(), dataSourceMetadata.getAllowedRoles().toString()),
+            RestStatus.UNAUTHORIZED);
```

Comment:
should this be 403 instead of 401?

Reply:
Yeah you are right. This is a backport PR. will address this in a new PR.